src/main/java/org/broadinstitute/hellbender/utils/codecs/sampileup/SAMPileupCodec.java

```diff
@@ -0,0 +1,343 @@
+ *
+ *
+ * @author Matt Hanna, Geraldine VdAuwera
+ * @since 2014
+ */
+public class SAMPileupCodec extends AsciiFeatureCodec<SAMPileupFeature> {
+    // number of tokens expected (6 or 10 are valid, anything else is wrong)
+    private static final int basicTokenCount = 6;
+    private static final int consensusSNPTokenCount = 10;
+    private static final int consensusIndelTokenCount = 13;
+    private static final char fldDelim = '\t';
+    // allocate once and don't ever bother creating them again:
+    private static final String baseA = "A";
+    private static final String baseC = "C";
+    private static final String baseG = "G";
+    private static final String baseT = "T";
+    private static final String emptyStr = ""; // we will use this for "reference" allele in insertions
+
+    // codec file extension
+    protected static final String FILE_EXT = "samp";
+
+    public SAMPileupCodec() {
+        super(SAMPileupFeature.class);
+    }
+
+    public SAMPileupFeature decode(String line) {
+        //+1 because we want to know if we have more than the max
+        String[] tokens = new String[consensusIndelTokenCount+1];
+
+        // split the line
+        final int count = ParsingUtils.split(line,tokens,fldDelim);
+
+        SAMPileupFeature feature = new SAMPileupFeature();
+
+        /**
+         * Tokens 0, 1, 2 are the same for both formats so they will be interpreted without differentiation.
+         * The 10/13-format has 4 tokens inserted after token 2 compared to the 6-format, plus 3 more tokens added at
+         * the end for indels. We are currently not making any use of the extra indel tokens.
+         *
+         * Any token count other than basicTokenCount, consensusSNPTokenCount or consensusIndelTokenCount is wrong.
+         */
+        final String observedString, bases, quals;
+
+        feature.setChr(tokens[0]);
+        feature.setStart(Integer.parseInt(tokens[1]));
+
+        if(tokens[2].length() != 1)  {
+            throw new CodecLineParsingException("The SAM pileup line had unexpected base " + tokens[2] + " on line = " + line);
+        }
+        feature.setRef(tokens[2].charAt(0));
+
+        switch (count) {
+            case basicTokenCount:
+                bases = tokens[4];
+                quals = tokens[5];
+                // parsing is pretty straightforward for 6-col format
+                if ( feature.getRef() == '*' ) {   // this indicates an indel -- but it shouldn't occur with vanilla 6-col format
+                    throw new CodecLineParsingException("Found an indel on line = " + line + " but it shouldn't happen in simple pileup format");
+                } else {
+                    parseBasesAndQuals(feature, bases, quals);
+                    feature.setRefBases(tokens[2].toUpperCase());
+                    feature.setEnd(feature.getStart());
+                }
+                break;
+            case consensusSNPTokenCount: // pileup called a SNP or a reference base
```

Comment:
unless you need them I would just remove the parsing for deprecated pileups and greatly simplify this class